packages/dashboard/src/pages/templates/email/[id].page.tsx

```diff
@@ -32,18 +32,31 @@ export const getServerSideProps: GetServerSideProps<PropsWithInitialState> =
     }
     const workspaceId = dfContext.workspace.id;
 
-    const [emailTemplate, userProperties] = await Promise.all([
-      prisma().messageTemplate.findUnique({
-        where: {
-          id: templateId,
-        },
-      }),
-      prisma().userProperty.findMany({
-        where: {
-          workspaceId,
-        },
-      }),
-    ]);
+    const [emailTemplate, userProperties, defaultEmailProvider] =
+      await Promise.all([
+        prisma().messageTemplate.findUnique({
+          where: {
+            id: templateId,
+          },
+        }),
+        prisma().userProperty.findMany({
+          where: {
+            workspaceId,
+          },
+        }),
+        prisma().defaultEmailProvider.findUnique({
+          where: {
+            workspaceId,
+          },
+        }),
+      ]);
+
+    const definition = defaultEmailDefinition();
+    if (defaultEmailProvider?.fromAddress)
+      definition.from = definition.from.replace(
```

Comment:
Hey @tobihans could you explain this code? Why are you replacing `/default:\s+".*"/` with `default: "${defaultEmailProvider.fromAddress}" `?

Instead, don't we want to modify the `defaultEmailDefinition` function to take the email provider as an argument and use the from value if there is one?

Reply:
You're right. I've updated it. Thank you. I wanted to only replace `hello@company.com` with the default email.